setup.cfg

```diff
@@ -30,11 +30,6 @@ install_requires =
     numpy>=1.16.6
     scipy>=1.0
     packaging
-setup_requires =
-    numpy>=1.16.6,<1.20
-    scipy>=1.0
-    cython>=0.29.20
-    packaging
```

Comment:
Removing this prevents use of `python setup.py develop` and I don't think there is another way to rebuild only the Cython files that have changed while developing at the moment. :/

Reply:
This has been deprecated for a while now, and recently has started producing warnings. I would recommend requiring users running `setup.py` to pre-install requirements. And I'd worry about using this command: https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html 

(But removing isn't important for this PR)

Reply:
Put it back in, but without the problematic numpy constraints.

Reply:
Thanks!

Re the article -- yes, I'm aware of the various pip options, but `pip install -e .` doesn't cache Cython file builds (maybe there is some new intree-builds version of pip that I haven't seen yet that does?).

Reply:
Have you tried this with pip 21.3? I'd have thought this might have started working now. Haven't tested it yet for this, though.

Reply:
It looks like it does with pip 21.3 if I run with `pip install -e . --no-build-isolation -v`. Not the friendliest command, but progress?